mobile/mysterium/wireguard_connection_setup.go

```diff
@@ -120,17 +121,33 @@ func (c *wireguardConnection) Start(options connection.ConnectOptions) (err erro
 		}
 	}()
 
-	if config.LocalPort > 0 {
-		err = c.natPinger.PingProvider(
+	if config.LocalPort > 0 || len(config.Ports) > 0 {
+		conn, err := c.natPinger.PingProvider(
 			config.Provider.Endpoint.IP.String(),
-			config.RemotePort,
-			config.LocalPort,
+			c.ports,
+			config.Ports,
 			0,
-			c.pingerStop,
 		)
 		if err != nil {
 			return errors.Wrap(err, "could not ping provider")
 		}
+
+		if conn != nil {
```

Comment:
Is this check for conn != nil is needed? I would expect conn to be defined if err is nil.

Reply:
Added this because of the test which uses mocks which return `nil` connection and `nil` error.

Reply:
Looks like mock is not correct.

Reply:
Fixed it.